lib/admin/WP_Auth0_Admin_Features.php

```diff
@@ -423,19 +423,24 @@ public function security_validation( $old_options, $input ) {
 					__( 'No database connections found for this application. ', 'wp-auth0' ) .
 					$this->get_dashboard_link( 'connections/database', __( 'See all database connections', 'wp-auth0' ) )
 				);
+				$input['password_policy'] = $old_options['password_policy'];
+				return $input;
 			}
 
 			foreach ( $connections as $connection ) {
 				if ( in_array( $input['client_id'], $connection->enabled_clients ) ) {
-					$patch       = array( 'options' => array( 'passwordPolicy' => $input['password_policy'] ) );
-					$update_resp = WP_Auth0_Api_Client::update_connection( $domain, $app_token, $connection->id, $patch );
+					$u_connection                          = clone $connection;
+					$u_connection->options->passwordPolicy = $input['password_policy'];
+
+					$update_resp = WP_Auth0_Api_Client::update_connection( $domain, $app_token, $u_connection->id, $u_connection );
 
 					if ( false === $update_resp ) {
 						$this->add_validation_error(
 							__( 'There was a problem updating the password policy. ', 'wp-auth0' ) .
 							__( 'Please manually review and update the policy. ', 'wp-auth0' ) .
 							$this->get_dashboard_link( 'connections/database', __( 'See all database connections', 'wp-auth0' ) )
 						);
+						$input['password_policy'] = $old_options['password_policy'];
```

Comment:
I don't see where in the code you're saving the "new value" *before* doing the request, but this line seems to add nothing. If anything, the `update_connection` method should NOT modify options unless the request was successful (as part of the method logic). Why would it be able to modify the options if it hasn't checked the request results first?

Reply:
This method returns `$input`, which contains the new values. This reverts this specific value to the value it was before the admin change. API call failed so nothing should update.